### PR TITLE
Add `work_in_progress` enum

### PR DESCRIPTION
**Pull request by jessieay**
* Submitting delivery URL sets auction to `work_in_progress`
* Closes https://github.com/18F/micropurchase/issues/1274

### CODE REVIEW

File: app/models/auction.rb [hunk 1]
```diff
@@ -54,10 +55,6 @@ class Auction < ActiveRecord::Base
   )
   validate :publishing_auction, on: :update, if: :published_changed?
 
-  def work_in_progress?
-    delivery_url.present? && pending_delivery?
-  end
-
```
Comment (harrisj) [inline app/models/auction.rb]: Ooooh

File: spec/factories/auctions.rb [hunk 2]
```diff
@@ -108,6 +108,12 @@
       delivery_status :accepted_pending_payment_url
     end
 
+    trait :work_in_progress do
+      closed
+      with_bids
+      delivery_status :work_in_progress
```
Comment (harrisj) [inline spec/factories/auctions.rb]: Do you want to give this trait a delivery_url too?

Comment (jessieay) [inline spec/factories/auctions.rb]: good idea!

Comment (jessieay) [inline spec/factories/auctions.rb]: added at 2806859

**Comment by harrisj** [comment]
LGTM!

**Pull request merged**